app/utils/render-component.js

```diff
@@ -46,14 +46,28 @@ export function getRenderComponentFor(parent) {
 	};
 }
 
-export function queryPlaceholders($element) {
+function isChildOfHidden(element) {
+	let node = element.parentElement;
+	while (node != null) {
+		if (node.classList.contains('mobile-hidden')) {
+			return true;
+		}
+		node = node.parentElement;
+	}
+	return false;
+}
+
+export function queryPlaceholders(element, filterHidden) {
 	const components = [];
+	let componentElements = element.querySelectorAll('[data-component]');
 
-	$element.find('[data-component]').each(function () {
-		const name = this.getAttribute('data-component'),
-			attrs = componentAttributes(this);
+	componentElements.forEach((componentElement) => {
```

Comment:
What do you think about using https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/offsetParent
this should return null if element is not visible
something like this:
[...componentElements].filter((componentElement) => componentElement.parentOffset)